### PR TITLE
CableNetApp/Porting OMP Atomic to AtomicUtilities

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
Porting OMP Atomic statements to use Atomic Utilities

- Atomic Utilities
- CableNetApplication

**Changelog**
- Changed statements from empirical_spring.cpp
- Changed statements from ring_element_3D.cpp
- Changed statements from sliding_cable_element_3D.cpp
- Changed statements from weak_coupling_slides.cpp

### CODE REVIEW

File: applications/CableNetApplication/custom_elements/empirical_spring.cpp [hunk 1]
```diff
@@ -540,12 +539,10 @@ void EmpiricalSpringElement3D2N::AddExplicitContribution(
...
             int index = i * msDimension;
 
-            #pragma omp atomic
-            r_nodal_mass += mass_vector[index];
+            AtomicAdd(r_nodal_mass, mass_vector[index]);
 
             for (int k = 0; k < msDimension; ++k) {
-                #pragma omp atomic
-                r_nodal_inertia[k] += 0.0;
+                AtomicAdd(r_nodal_inertia[k], 0.0);
```
Comment (ashishdarekar) [inline applications/CableNetApplication/custom_elements/empirical_spring.cpp]: I think we don't need this loop.

File: applications/CableNetApplication/custom_elements/ring_element_3D.cpp [hunk 2]
```diff
@@ -625,12 +624,10 @@ void RingElement3D::AddExplicitContribution(
...
             int index = i * dimension;
 
-            #pragma omp atomic
-            r_nodal_mass += mass_vector[index];
+            AtomicAdd(r_nodal_mass, mass_vector[index]);
 
             for (int k = 0; k < dimension; ++k) {
-                #pragma omp atomic
-                r_nodal_inertia[k] += 0.0;
+                AtomicAdd(r_nodal_inertia[k], 0.0);
```
Comment (ashishdarekar) [inline applications/CableNetApplication/custom_elements/ring_element_3D.cpp]: Same

File: applications/CableNetApplication/custom_elements/sliding_cable_element_3D.cpp [hunk 3]
```diff
@@ -752,12 +751,10 @@ void SlidingCableElement3D::AddExplicitContribution(
...
             int index = i * dimension;
 
-            #pragma omp atomic
-            r_nodal_mass += mass_vector[index];
+            AtomicAdd(r_nodal_mass, mass_vector[index]);
 
             for (int k = 0; k < dimension; ++k) {
-                #pragma omp atomic
-                r_nodal_inertia[k] += 0.0;
+                AtomicAdd(r_nodal_inertia[k], 0.0);
```
Comment (ashishdarekar) [inline applications/CableNetApplication/custom_elements/sliding_cable_element_3D.cpp]: same

**Comment by KlausBSautter** [comment]
nice :+1: , I'll check my local tests and let you know. Did you run the cable net tests in full debug? @ashishdarekar

**Comment by ashishdarekar** [comment]
> nice  , I'll check my local tests and let you know. Did you run the cable net tests in full debug? @ashishdarekar

@KlausBSautter Yes, I have run applications/CableNetApplication/tests/test_CableNetApplication.py

**Review by KlausBSautter — APPROVED** [review]

**Pull request merged**